tests/kafkatest/tests/streams/streams_upgrade_test.py

```diff
@@ -119,36 +113,37 @@ def test_upgrade_downgrade_brokers(self, from_version, to_version):
                    err_msg="Broker did not create all topics in 60 seconds ")
 
         self.driver = StreamsSmokeTestDriverService(self.test_context, self.kafka)
-        self.driver.disable_auto_terminate()
 
-        self.processor1 = StreamsSmokeTestJobRunnerService(self.test_context, self.kafka)
+        processor = StreamsSmokeTestJobRunnerService(self.test_context, self.kafka)
         
         self.driver.start()
-        self.processor1.start()
-
-        self.verify_process_records()
+        with processor.node.account.monitor_log(processor.STDOUT_FILE) as monitor:
+            processor.start()
+            monitor.wait_until(self.processed_msg,
+                               timeout_sec=60,
+                               err_msg="Never saw output '%s' on" % self.processed_msg + str(processor.node))
 
         connected_message = "Discovered group coordinator"
-        with self.processor1.node.account.monitor_log(self.processor1.LOG_FILE) as log_monitor:
-            with self.processor1.node.account.monitor_log(self.processor1.STDOUT_FILE) as stdout_monitor:
+        with processor.node.account.monitor_log(processor.LOG_FILE) as log_monitor:
+            with processor.node.account.monitor_log(processor.STDOUT_FILE) as stdout_monitor:
 
                 self.perform_broker_upgrade(to_version)
 
                 log_monitor.wait_until(connected_message,
                                        timeout_sec=120,
-                                       err_msg=("Never saw output '%s' on " % connected_message) + str(self.processor1.node.account))
+                                       err_msg=("Never saw output '%s' on " % connected_message) + str(processor.node.account))
 
                 stdout_monitor.wait_until(self.processed_msg,
                                           timeout_sec=60,
-                                          err_msg="Never saw output '%s' on" % self.processed_msg + str(self.processor1.node.account))
-
+                                          err_msg="Never saw output '%s' on" % self.processed_msg + str(processor.node.account))
+        self.driver.wait()
         self.driver.stop()
```

Comment:
Doesn't `wait()` mean, block until terminated? If yes, no need to cal `stop()` afterwards (would be a no-op)?

Reply:
This was pre-existing in the test, also used this way in `streams_smoke_test.py`

Reply:
Hmm.. I think it should be 

```
self.driver.stop()
self.driver.wait()
```

instead as used in other places.

Not sure why the test itself did not fail though, without calling `stop()` the `wait()` call should fail.